subsys/net/lib/lwm2m_client_utils/location/location_assistance.c

```diff
@@ -30,9 +31,112 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #define AGPS_LOCATION_PATHS_DEFAULT 7
 #define AGPS_LOCATION_PATHS_MAX 8
 
+static struct lwm2m_ctx *req_client_ctx;
+static bool req_confirmable;
+
+#define INITIAL_RETRY_INTERVAL 675
+#define MAXIMUM_RETRY_INTERVAL 86400
```

Comment:
The maximum retry interval is 24 hours, which seems like a very long time. Also, this kind of automatic retry may not be desired in all cases.

With A-GPS, GNSS sends a new request for A-GPS data after one hour if it hasn't got what it asked for. The data requested by GNSS may change over time because more assistance data may have expired, so instead of repeating the same request again and again, it would make sense to always use the latest A-GPS data request from GNSS.

Depending on the use case, retrying the A-GPS download may be undesirable. The device might for example try to get one GNSS fix a week. If A-GPS data can't be downloaded before the fix is got, trying to download the A-GPS data after this is just waste of energy.

With cellular location, if sending the location succeeds after several hours, the location is several hours old when it reaches the cloud. This might be fine if the data has a timestamp when the cell measurements were made, but if the data has no timestamp, there's no way of knowing whether that location is valid or not. If the device is moving, it will not be valid.

Instead of automatic retries, I think the application should be able to decide what to do.

Reply:
Fair points.
We are not buffering the old requests somewhere but the current data in the lwm2m-objects will be used.
For A-GPS, it is a small change to allow setting the mask, I'll do that, connmon is updated regardless. For P-GPS, no changes needed.

For cellular data, connmon and the neighboring cells are updated all the time regardless of the location assistance objects, so it is having latest data with every send.

Actually application can decide whether to use the automatic retries or not. If application doesn't call `location_assistance_init` ever, the automatic retries aren't used. The function should be renamed though.

Reply:
Thanks for the clarification. Yes, I think it would be good to the let the application decide whether automatic retries are used or not.

Reply:
I renamed the `location_assistance_init` to `location_assistance_init_resend_handler` and made doxygen changes.